### PR TITLE
Add `Client::get_slot_index`

### LINKED ISSUE #1123: Implement `Client::get_slot_index`
It's a TODO atm

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Adds `Client::get_slot_index` based on the old functionality of `get_time_checked`.

## Links to any relevant issues

Closes #1123 
Closes https://github.com/iotaledger/iota-sdk/issues/1073

## Open Questions

- Do we want to change the checking at all?
- Should we even cache the tangle time?
- Why is the accepted tangle time an optional field?

### CODE REVIEW

File: sdk/src/client/api/high_level.rs [hunk 1]
```diff
@@ -115,15 +115,14 @@ impl Client {
...
-        let current_time = unix_timestamp_now().as_secs() as u32;
+    // Returns the slot index corresponding to the current timestamp.
+    pub async fn get_slot_index(&self) -> Result<SlotIndex> {
+        let current_time = unix_timestamp_now().as_secs();
 
         let network_info = self.get_network_info().await?;
 
         if let Some(tangle_time) = network_info.tangle_time {
-            let tangle_time = tangle_time as u32;
+            let tangle_time = tangle_time;
```
Comment (kwek20) [inline sdk/src/client/api/high_level.rs]: :D

Comment (DaughterOfMars) [reply sdk/src/client/api/high_level.rs]: hahaha oops

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/src/client/api/high_level.rs [hunk 2]
```diff
@@ -133,11 +131,6 @@ impl Client {
             }
         }
 
-        Ok(current_time)
-    }
-
-    // TODO
-    pub async fn get_slot_index(&self) -> Result<SlotIndex> {
-        todo!()
+        Ok(self.get_protocol_parameters().await?.slot_index(current_time))
```
Comment (thibault-martinez) [inline sdk/src/client/api/high_level.rs]: Maybe reuse `network_info`? This would do an additional call on wasm

Comment (DaughterOfMars) [reply sdk/src/client/api/high_level.rs]: Ah, good call

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Review by abdulmth — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**